packages/core/src/components/button/buttons.tsx

```diff
@@ -41,41 +29,50 @@ export class Button extends React.Component<React.HTMLProps<HTMLButtonElement> &
     public static displayName = "Blueprint.Button";
 
     public render() {
-        return (
-            <button
-                type="button"
-                {...removeNonHTMLProps(this.props)}
-                className={getButtonClasses(this.props)}
-                ref={this.props.elementRef}
-            >
-                {this.props.text}
-                {this.props.children}
-                {maybeRenderRightIcon(this.props.rightIconName)}
-            </button>
-        );
+        return <button type="button" {...getButtonHTMLProps(this.props)} {...getButtonProps(this.props)} />;
     }
 }
 
 export class AnchorButton extends React.Component<React.HTMLProps<HTMLAnchorElement> & IButtonProps, {}> {
     public static displayName = "Blueprint.AnchorButton";
 
     public render() {
-        const { children, disabled, href, onClick, rightIconName, text } = this.props;
-        return (
-            <a
-                {...removeNonHTMLProps(this.props)}
-                className={getButtonClasses(this.props)}
-                href={disabled ? null : href}
-                onClick={disabled ? null : onClick}
-                ref={this.props.elementRef}
-                role="button"
-                tabIndex={disabled ? null : 0}
-            >
-                {text}
-                {children}
-                {maybeRenderRightIcon(rightIconName)}
-            </a>
-        );
+        return <a role="button" {...getButtonHTMLProps(this.props)} {...getButtonProps(this.props)} />;
+    }
+}
+
+function getButtonHTMLProps(props: IButtonProps) {
+    return removeNonHTMLProps(props, props.disabled ? DISABLED_PROPS : [], true);
+}
+
+// spread this after HTML props to override `className`
```

Comment:
I don't actually think this refactor helps much, it instead reduces readability by introducing more indirection. On top of that you have to know which props-generating function result to apply first as props, _and_ `children` are supplied with a nonstandard syntax.